dom.bs

```diff
@@ -5875,19 +5875,19 @@ dictionary ShadowRootInit {
 initialized.
 
 <p>An <a for=/>element</a>'s <a for=Element>custom element state</a> is one of
-"<code>undefined</code>", "<code>failed</code>", "<code>uncustomized</code>", or
-"<code>custom</code>". An <a for=/>element</a> whose <a for=Element>custom element state</a> is
-"<code>uncustomized</code>" or "<code>custom</code>" is said to be
-<dfn export id=concept-element-defined for=Element>defined</dfn>. An <a for=/>element</a> whose
-<a for=Element>custom element state</a> is "<code>custom</code>" is said to be
-<dfn export id=concept-element-custom for=Element>custom</dfn>.
+"<code>undefined</code>", "<code>failed</code>", "<code>uncustomized</code>",
+"<code>precustomized</code>", or "<code>custom</code>". An <a for=/>element</a> whose <a
+for=Element>custom element state</a> is "<code>uncustomized</code>", "<code>precustomized</code>",
```

Comment:
This seems wrong. Why would a custom element be considered as *defined* before `super()` call is made in the constructor? This will be incompatible with the shipping behavior of custom elements.

Reply:
I think I agree. I've removed "precustomized" from the definition of "defined".

@annevk do you still think there is a need for more testing of `:defined`?

Reply:
I think we should have a test yes, to ensure that it does not match that.

Reply:
Ok, I'll add a test that `:defined` doesn't match, pre-`super()`.

Reply:
So we already have [a test for :defined](https://github.com/web-platform-tests/wpt/blob/master/custom-elements/pseudo-class-defined.html#L103) during an upgrade, and I think it's good enough to catch problems here. Note a couple things:
 * `:defined` already doesn't match during an upgrade, even **after** `super()`, because the custom element state is only set to "custom" [after the constructor completes](https://html.spec.whatwg.org/#upgrades).
 * Testing `:defined` prior to `super()` shouldn't be typical anyway, since you can't access `this` prior to `super()`. You could use an external reference to the element to test, of course, but that doesn't seem like it'd be common.

I'm sure I'm missing something here - please let me know what it is, and what you'd like me to test for. I can add a test for the second bullet above, but it doesn't seem terribly useful. It would catch the case where `:defined` matches **prior** to `super()` but stops matching it **after** `super()`.

Reply:
I added [this additional testing](https://github.com/web-platform-tests/wpt/pull/25794/files), in case that's what you're looking for here.

Reply:
Thanks, I think those asserts are helpful.